### PR TITLE
Feature#1 Database design and configuration

### LINKED ISSUE #1: Design and Configuration of the database on Firebase
# REQUIREMENTS
We need to have the database that will be used in this project designed. The proposal may change depending on the needs that arise.

# Acceptance Criteria
List all required acceptance criteria
- [ ] Connection with firebase

### PR DESCRIPTION
**Pull request by Alonso-AS**
- [x] The database design was made in firebase.
- [x] The user authentication option was enabled in firebase.
- [x] The application "citizen complaint" was configured as indicated by the official firebase website.
- [x] Test code was made to identify the connection between firebase and the app.
- [x] This test code will be left only as base code for issues where it can be implemented.

### **Test**

To test the application, access the DenunciaCiudadana folder, Install Node dependencies through the following command:

`npm install`

To run the application use the following commands:

**Linux**

`npx react-native start`

`npx react-native run android`

**Windows**

`npx react-native run android`

### **Desing**

image
image

### CODE REVIEW

**Review by jkzasori — DISMISSED** [review]
I just recommend to write the names of the functions and variables in English because it's easy for people from everywhere to read your code.

File: DenunciaCiudadana/App.js [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
-  Text,
-  View,
-} from 'react-native';
+import React, {useState, useEffect} from 'react';
+import auth from '@react-native-firebase/auth';
+import Login from './src/Login';
+import SignUp from './src/SignUp';
 
+export default function App() {
+  // Set an initializing state whilst Firebase connects
```
Comment (fernandoamz) [inline DenunciaCiudadana/App.js]: remove this comment

File: DenunciaCiudadana/src/Login.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      .add({
+        title: 'Funcionario público es visto comprando votos',
+        description: 'Un funcionario público fue visto comprando votos',
+        area: 'Goverment',
+        like: 0,
+        signedBy: true,
+        user: 'users/' + user.uid,
+        dateEvent: Date.now(),
+        timestamp: Date.now(),
+      })
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/Login.js]: Why these values are hardcoded? 🤔

File: DenunciaCiudadana/src/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import {View, Text, Button} from 'react-native';
+import auth from '@react-native-firebase/auth';
+
+const SignUp = () => {
+  const crearUsuario = () => {
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/SignUp.js]: The function's name and variable's name doesn't name in spanish

File: DenunciaCiudadana/src/SignUp.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        console.error(error);
+      });
+  };
+
+  return (
+    <View>
+      <Text>NAME</Text>
+      <Text>password</Text>
+      <Button title="entrar" onPress={crearUsuario} />
+      <Button title="ya tengo Email" onPress={yaEstoyLogeado} />
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/SignUp.js]: Add all these strings inside constant file

**Review by fernandoamz — CHANGES_REQUESTED** [review]
Clean a little bit more the code

**Review by jkzasori — APPROVED** [review]

File: DenunciaCiudadana/src/Login.js (hunk 2 above)
Comment (DiegoSpielzeug) [reply DenunciaCiudadana/src/Login.js]: It is just to make sure thath the db is working, but in the future we are going to use redux for that oparation

File: DenunciaCiudadana/src/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        console.error(error);
+      });
+  };
+
+  return (
+    <View>
+      <Button title={SIGN_IN} onPress={createUser} />
+      <Button title={LOG_IN} onPress={loggedIn} />
+    </View>
```
Comment (LeyaimMagma) [inline DenunciaCiudadana/src/SignUp.js:51]: Could you refactor this?

**Review by LeyaimMagma — CHANGES_REQUESTED** [review]

**Pull request merged**